src/test/groovy/org/prebid/server/functional/tests/DsaSpec.groovy

```diff
@@ -0,0 +1,263 @@
+package org.prebid.server.functional.tests
+
+import org.prebid.server.functional.model.db.StoredRequest
+import org.prebid.server.functional.model.request.amp.AmpRequest
+import org.prebid.server.functional.model.request.auction.BidRequest
+import org.prebid.server.functional.model.request.auction.RegsDsa
+import org.prebid.server.functional.model.response.auction.BidExt
+import org.prebid.server.functional.model.response.auction.BidExtDsa
+import org.prebid.server.functional.model.response.auction.BidResponse
+
+import static org.prebid.server.functional.model.response.auction.ErrorType.GENERIC
+
+class DsaSpec extends BaseSpec {
+
+    def "PBS should populate AMP bid request with DSA regs and send DSA to bidder and succeed when DSA is not required"() {
+        given: "Default AmpRequest"
+        def ampRequest = AmpRequest.defaultAmpRequest
+        def dsa = RegsDsa.getDefaultRegsDsa(dsaRequired)
+
+        and: "Stored default bid request with DSA"
+        def ampStoredRequest = BidRequest.defaultBidRequest.tap {
+            regs.ext.dsa = dsa
+            setAccountId(ampRequest.account)
+        }
+
+        and: "Save storedRequest into DB"
+        def storedRequest = StoredRequest.getStoredRequest(ampRequest, ampStoredRequest)
+        storedRequestDao.save(storedRequest)
+
+        and: "Default bidder response with DSA"
+        def bidResponse = BidResponse.getDefaultBidResponse(ampStoredRequest).tap {
+            seatbid[0].bid[0].ext = bidExt
+        }
+
+        and: "Set bidder response"
+        bidder.setResponse(ampStoredRequest.id, bidResponse)
+
+        when: "PBS processes amp request"
+        defaultPbsService.sendAmpRequest(ampRequest)
+
+        then: "Bidder request should contain DSA"
+        def bidderRequests = bidder.getBidderRequest(ampStoredRequest.id)
+        verifyAll {
+            bidderRequests.regs.ext.dsa.required == dsa.required
+            bidderRequests.regs.ext.dsa.datatopub == dsa.datatopub
+            bidderRequests.regs.ext.dsa.pubrender == dsa.pubrender
+            bidderRequests.regs.ext.dsa.transparency[0].domain == dsa.transparency[0].domain
+            bidderRequests.regs.ext.dsa.transparency[0].params == dsa.transparency[0].params
+        }
+
+        where:
+        dsaRequired || bidExt
+        0           || null
+        1           || new BidExt("dsa": null)
+    }
+
+    def "PBS should populate AMP bid request with DSA regs and send DSA to bidder and always succeed when bidder returns DSA"() {
```

Comment:
1. PBS amp should return seatbid[].bid[].ext.dsa when DSA is not empty and required is #dsaRequired
2. Please check seat.bid.ext.dsa also

Reply:
As I understand AMP response doesn't contain this info